### PR TITLE
Add blog post for UUID4::Tiny

### PR DESCRIPTION
**Pull request by m-dango**

### CODE REVIEW

File: _posts/2019-09-18-uuid4-tiny.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+author: Daniel Mita
+layout: post
+title: UUID4::Tiny
+subtitle: Cryptographically secure version 4 UUIDs via the Linux getrandom() system call
+tags: perl uuid cryptography cpan module
+---
+
+[UUID::Tiny][uuid-tiny] is nice, but unfortunately due to its usage of
+rand() the version 4 UUIDs it provides are not cryptographically
+strong. We opted to create a lightweight (core modules only!)
```
Comment (colinnewell) [inline _posts/2019-09-18-uuid4-tiny.md]: we can not promise that the next uuid we generate won't be guessable might be a clearer way to explain it?

File: _posts/2019-09-18-uuid4-tiny.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+alternative which uses a strong source of randomness.
+
+Linux kernel v3.17 introduced the system call getrandom() which
+returns a specified number of random bytes. It reads from the same
+source as /dev/urandom by default without requiring the opening of
+a file handle.
```
Comment (colinnewell) [inline _posts/2019-09-18-uuid4-tiny.md]: the lack of need for a file handle removes another potential attack against the gathering of the random numbers.  Might be worth sneaking that in there to explain why this is worth doing.

Comment (m-dango) [reply _posts/2019-09-18-uuid4-tiny.md]: I don't know much about how this attack works, is there something I can reference?

Comment (colinnewell) [reply _posts/2019-09-18-uuid4-tiny.md]: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=c6e9d6f38894798696f23c8084ca7edbf16ee895

The bit about "resiliance against
file descriptor exhaustion attacks" is the thing we're talking about.

File: _posts/2019-09-19-uuid4-tiny.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+rand() the version 4 UUIDs it provides are not cryptographically
+strong i.e. there exists the potential for the next value to be
+predicted. We opted to create a lightweight (core modules only!)
+alternative which uses a strong source of randomness.
+
+Linux kernel v3.17 introduced the system call getrandom() which
+returns a specified number of random bytes. It reads from the same
+source as /dev/urandom by default, without requiring the opening of
+a file handle (which removes an avenue for file descriptor
+exhaustion attacks).
```
Comment (colinnewell) [inline _posts/2019-09-19-uuid4-tiny.md]: It's kind of the other way around.  It's that file descriptor exhaustion attacks could impede our ability to get reliable random numbers by preventing us from being able to open /dev/urandom in the first place.  By using a syscall we don't have that worry.

**Pull request merged**